Demos/Shopping/ShoppingUITests/TestPages/CartPage.swift

```diff
@@ -0,0 +1,88 @@
+    let app: XCUIApplication
+    var parentView: ParentView
+    
+    init(app: XCUIApplication, parentView: ParentView, file: StaticString = #file, line: UInt = #line) {
+        self.app = app
+        self.parentView = parentView
+        XCTAssertTrue(app.collectionViews.firstMatch.waitForExistence(), "Cart didn't finish loading", file: file, line: line)
+    }
+    
+    @discardableResult
+    func assertEmptyCart(file: StaticString = #file, line: UInt = #line) -> Self {
+        assert(app.staticTexts["Your cart is empty."].waitForExistence(), message: "Cart is not empty", file: file, line: line)
+    }
+    
+    @discardableResult
+    func assertReceiptExists(file: StaticString = #file, line: UInt = #line) -> Self {
+        assert(app.staticTexts["Receipt"].waitForExistence(), message: "Can't find 'Receipt' view", file: file, line: line)
+    }
+    
+    @discardableResult
+    func assertTotal(price: String, file: StaticString = #file, line: UInt = #line) -> Self {
+        assert(app.staticTexts["Total: \(price)"].waitForExistence(), message: "Can't find 'Total: \(price)' text", file: file, line: line)
+    }
+    
+    @discardableResult
+    func assertInsufficientFunds(file: StaticString = #file, line: UInt = #line) -> Self {
+        assert(app.staticTexts["Insufficient funds!"].waitForExistence(), message: "Can't find 'Insufficient funds!'", file: file, line: line)
+    }
+    
+    @discardableResult
+    func assertRowExists(for product: TestProduct, file: StaticString = #file, line: UInt = #line) -> Self {
+        let row = app.collectionViews.cells.containing(.staticText, identifier: product.name).firstMatch
+        return assert(row.waitForExistence(), message: "Can't find row with '\(product.name)' text", file: file, line: line)
+            .assert(row.staticTexts[product.price].exists, message: "Can't find '\(product.price)' text in row", file: file, line: line)
+    }
+        
+    @discardableResult
+    func removeRow(for product: TestProduct, file: StaticString = #file, line: UInt = #line) -> Self {
+        let row = app.collectionViews.cells.containing(.staticText, identifier: product.name).firstMatch
+        let deleteButton = app.buttons["Remove \(product.name)"]
+        XCTAssertTrue(row.waitForExistence(), "Can't find '\(product.name) Row'", file: file, line: line)
+        row.swipeLeft()
+        XCTAssertTrue(deleteButton.waitForExistence(), "Can't find 'Remove \(product.name)'", file: file, line: line)
+        deleteButton.tap()
+        XCTAssertTrue(app.activityIndicators["Processing..."].waitForNonexistence(), "'Processing...' is stuck", file: file, line: line)
+        return self
+    }
+    
+    @discardableResult
+    func tapPlaceOrder(file: StaticString = #file, line: UInt = #line) -> Self {
+        XCTAssertTrue(app.buttons["Place Order"].waitForExistence(), "Can't find 'Place Order' button", file: file, line: line)
+        app.buttons["Place Order"].tap()
```

Comment:
`lazy var placeOrderButton = app.buttons["Place Order"]` ? might be better?

Reply:
Good call. @vhtisdale made a similar recommendation to improve readability and reduce number of places where changes have to be made.

Although, `lazy` requires mutability, so I opted for calculated properties. Should still be very performant considering that XCTest doesn't evaluate the element query until you operate on it or examine its values.